tests/exponential.rs

```diff
@@ -0,0 +1,380 @@
+/*!
+    Exponential functions from arithmetic
+!*/
+
+use num::rational::Ratio;
+use num::BigInt;
+use ruler::*;
+
+pub type Rational = Ratio<BigInt>;
+
+egg::define_language! {
+    pub enum Exponential {
+        // trig operators
+        "exp" = Exp(Id),
+        "log" = Log(Id),
+        "pow" = Pow([Id; 2]),
+        "sqrt" = Sqrt(Id),
+        "cbrt" = Cbrt(Id),
+
+        // arithmetic operators
+        "~" = Neg(Id),
+        "+" = Add([Id; 2]),
+        "-" = Sub([Id; 2]),
+        "*" = Mul([Id; 2]),
+        "/" = Div([Id; 2]),
+
+        // constants
+        Num(Rational),
+        Var(Symbol),
+    }
+}
+
+impl SynthLanguage for Exponential {
+    type Constant = Rational;
+
+    // cvec-less domain
+    fn eval<'a, F>(&'a self, _cvec_len: usize, _get_cvec: F) -> CVec<Self>
+    where
+        F: FnMut(&'a Id) -> &'a CVec<Self>,
+    {
+        vec![]
+    }
+
+    fn initialize_vars(egraph: &mut EGraph<Self, SynthAnalysis>, vars: &[String]) {
+        for v in vars {
+            let id = egraph.add(Exponential::Var(Symbol::from(v.clone())));
+
+            let l_id = egraph.add(Exponential::Log(id));
```

Comment:
Why are you adding these here? shouldn't these be in the workload?

Reply:
Oh I forgot to union those expressions! Is there a better way to assert that `?x <==> (exp (log ?x))`  (and `?x <==> (log (exp ?x))` only for variables?